cli/internal/core/engine.go

```diff
@@ -454,12 +454,12 @@ func (e *Engine) AddDep(fromTaskID string, toTaskID string) error {
 // ValidatePersistentDependencies checks if any task dependsOn persistent tasks and throws
 // an error if that task is actually implemented
 func (e *Engine) ValidatePersistentDependencies(graph *graph.CompleteGraph, concurrency int) error {
-	var validationError error
+	var validationErrors []string
 	persistentCount := 0
 
 	// Adding in a lock because otherwise walking the graph can introduce a data race
 	// (reproducible with `go test -race`)
-	var sema = util.NewSemaphore(1)
+	var mu sync.Mutex
```

Comment:
This wasn't strictly necessary, but we really just want a mutex guarding the validation errors and not a semaphore.